### PR TITLE
Issue #1602: Added two channels to binding

### PR DESCRIPTION
**Pull request by MHerbst**
Added two channels to show additional information for input type Net Radio.

Added descriptions to the README.md and corrected some spelling and formatting errors.

### CODE REVIEW

**Comment by kaikreuzer** [comment]
@MHerbst This seems to collide with https://github.com/openhab/openhab2-addons/pull/1341 - could you please check?

**Comment by MHerbst** [comment]
@kaikreuzer Yes there are some overlapping modifications. I will wait until #1341 was merged and then update this PR

**Comment by davidgraeff** [comment]
@MHerbst: I reviewed #1341 and some changes are necessary for merging. After the merge I will review your contribution again :)

File: addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/YamahaReceiverCommunication.java [hunk 1]
```diff
@@ -208,6 +213,26 @@ public void updateState(YamahaReceiverState state) throws IOException {
...
+                        "<?xml version=\"1.0\" encoding=\"utf-8\"?><YAMAHA_AV cmd=\"GET\">"
+                                + "<NET_RADIO><Play_Info>GetParam</Play_Info></NET_RADIO></YAMAHA_AV>");
+                Node nodeNetInfo = getNode(docNet.getFirstChild(), "NET_RADIO/Play_Info/Meta_Info");
+                node = getNode(nodeNetInfo, "Station");
+                state.netRadioStation = node != null ? node.getTextContent() : "";
+                node = getNode(nodeNetInfo, "Song");
+                state.netRadioSong = node != null ? node.getTextContent() : "";
+            } catch (IOException e) {
+                // Issue a warning if XML request is not supported by the receiver
+                logger.warn("Request to get additional Net Radio information failed: " + e.getMessage());
```
Comment (davidgraeff) [inline addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/YamahaReceiverCommunication.java]: The YamahaReceiverCommunication.java class has no logging on purpose. It is really just for sending/receiving network traffic. Just let the exception be handled by the calling class. If you want to add additional information (like your "Request to get additional Net Radio information failed: " string, handle the exception, and generate a new one but don't do any output here.

File: addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/YamahaReceiverCommunication.java [hunk 2]
```diff
@@ -208,6 +213,26 @@ public void updateState(YamahaReceiverState state) throws IOException {
         node = getNode(basicStatus, "Input/Input_Sel_Item_Info/Src_Number");
         value = node != null ? node.getTextContent() : "0";
         state.netRadioChannel = Integer.parseInt(value);
+
+        // Net Radio provides additional informaton like station name and currently played song
+        if ("NET RADIO".equals(state.input)) {
+            try {
+                Document docNet = postAndGetXmlResponse(
```
Comment (davidgraeff) [inline addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/YamahaReceiverCommunication.java]: The updateState() method should only cause one postAndGetXmlResponse call and no conditional other calls, for better testing. Please write an updateNetRadio() method that is called wherever necessary (Would be YamahaReceiverState.updateState() I think).

**Comment by MHerbst** [comment]
@davidgraeff I will wait for #1341 and then modify my additions accordingly. I the meantime I have seen that it is possible to get the RDS data for radio via Tuner. Therefore it might be better to change the channel name "netradiosong" to something more generic that can be used for Tuner and NetRadio.

**Comment by MHerbst** [comment]
Closed this pull request because of the large refactoring (see #1781). To me it seems that the modified add-on will contain the features that I have implemented here. Therefore the pull request is no longer necessary.

**Pull request closed without merging**